source/simulator/helper_functions.cc

```diff
@@ -827,24 +859,45 @@ namespace aspect
     if (parameters.use_locally_conservative_discretization)
       AssertThrow(false, ExcNotImplemented());
 
-    // TODO: currently does not work if velocity and
-    // pressure are in the same block.
-    Assert(introspection.block_indices.velocities != introspection.block_indices.pressure,
-           ExcNotImplemented());
-
     // In the following we integrate the right hand side. This integral is the
     // correction term that needs to be added to the pressure right hand side.
     // (so that the integral of right hand side is set to zero).
+    if (!parameters.include_melt_transport && introspection.block_indices.velocities != introspection.block_indices.pressure)
+      {
+        const double mean       = vector.block(introspection.block_indices.pressure).mean_value();
+        const double correction = (- mean * vector.block(introspection.block_indices.pressure).size()) / global_volume;
```

Comment:
This is the arithmetic mean of the vector elements, but this will not equal the "average pressure" (as computed by the integral of the pressure divided by the volume) in most cases (e.g., with cells of non-uniform size, as well as for DoFs at the boundary). Is this what you want?